neoemf-core/src/main/java/fr/inria/atlanmod/neoemf/bind/Bindings.java

```diff
@@ -0,0 +1,318 @@
+                    : type.newInstance();
+        }
+        catch (NoSuchMethodException | IllegalAccessException | InvocationTargetException | InstantiationException e) {
+            throw new BindingException(e);
+        }
+    }
+
+    // endregion
+
+    /**
+     * Retrieves the {@link URI} scheme for the speficied {@code type}.
+     * <p>
+     * The {@code type} must be a sub-class of {@link BackendFactory}, or must be annotated with {@link
+     * FactoryBinding}.
+     *
+     * @param type the type
+     *
+     * @return the {@link URI} scheme
+     *
+     * @throws BindingException if no scheme is defined for this {@code type}
+     */
+    @Nonnull
+    public static String schemeOf(Class<?> type) {
+        return UriBuilder.createScheme(nameOf(type));
+    }
+
+    /**
+     * Retrieves the {@link URI} scheme for the specified {@code factory}.
+     *
+     * @param factory the factory
+     *
+     * @return the {@link URI} scheme
+     *
+     * @throws BindingException if no scheme is defined for this {@code type}
+     */
+    @Nonnull
+    public static String schemeOf(BackendFactory factory) {
+        return UriBuilder.createScheme(factory.name());
+    }
+
+    /**
+     * Retrieves the name for the specified {@code type}.
+     * <p>
+     * The {@code type} must be a sub-class of {@link BackendFactory}, or must be annotated with {@link
+     * FactoryBinding}.
+     *
+     * @param type the type
+     *
+     * @return the name
+     *
+     * @throws BindingException if no name is defined for this {@code type}
+     */
+    @Nonnull
+    public static String nameOf(Class<?> type) {
+        return factoryFor(type).name();
+    }
+
+    /**
+     * Retrieves the {@link BackendFactory} associated to the {@code type}.
+     * <p>
+     * The {@code type} <b>must</b> be annotated with {@link FactoryBinding}.
+     *
+     * @param type the type of the instance to look for
+     *
+     * @return a new instance of {@link BackendFactory}
+     *
+     * @throws BindingException if no instance of {@link BackendFactory} is found for the {@code type}, or if an error
+     *                          occurs during the instantiation
+     */
+    @Nonnull
+    @SuppressWarnings("unchecked")
+    public static BackendFactory factoryFor(Class<?> type) {
+        checkNotNull(type, "type");
+
+        Class<? extends BackendFactory> factoryType = null;
+
+        if (BackendFactory.class.isAssignableFrom(type)) {
+            factoryType = (Class<? extends BackendFactory>) type;
+        }
+        else if (type.isAnnotationPresent(FactoryBinding.class)) {
+            factoryType = type.getAnnotation(FactoryBinding.class).value();
+        }
+
+        if (nonNull(factoryType)) {
+            return newInstance(factoryType);
+        }
+
+        throw new BindingException(
+                String.format("%s is not annotated with %s: Unable to retrieve the associated factory", type.getName(), FactoryBinding.class.getName()));
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') A "NullPointerException" could be thrown; "type" is nullable here. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2259)